docs/io/fits/usage/cloud.rst

```diff
@@ -0,0 +1,268 @@
+a secret access key in this case. You can use the ``fsspec_kwargs`` parameter
+to provide your key as follows:
+
+.. doctest-skip::
+
+    >>> fsspec_kwargs = {"key": "YOUR-SECRET-KEY-ID",
+    ...                  "secret": "YOUR-SECRET-KEY"}
+    >>> with fits.open(s3_uri, use_fsspec=True, fsspec_kwargs=fsspec_kwargs) as hdul:
+    ...     cutout = hdul[2].section[10:20, 30:50]
+
+.. warning::
+
+    Including secret access keys inside Python code is dangerous because you
+    may accidentally end up revealing your keys when you share your code with
+    others. A better practice is to store your access keys via a configuration
+    file or environment variables. See the ``s3fs`` documentation for guidance.
+
+
+Using :class:`~astropy.nddata.Cutout2D` with cloud-hosted FITS files
+====================================================================
+
+The examples above used the `ImageHDU.section` feature to download
+small cutouts given a set of pixel coordinates. For astronomical images it is
+often more convenient to obtain cutouts based on a sky position and angular
+size rather than array coordinates. For this reason, Astropy provides the
+`astropy.nddata.Cutout2D` tool which makes it easy to obtain cutouts informed
+by an image's World Coordinate System (`~astropy.wcs.WCS`).
+
+This cutout tool can be used in combination with ``fsspec`` and ``.section``.
+For example, assume you happen to know that the image we opened above contains
+a nice edge-on galaxy at the following position::
+
+    >>> # Approximate location of an edge-on galaxy
+    >>> from astropy.coordinates import SkyCoord
+    >>> position = SkyCoord('10h01m41.13s 02d25m20.58s')
+
+We also know that the radius of the galaxy is approximately 5 arcseconds::
+
+    >>> # Approximate size of the galaxy
+    >>> from astropy import units as u
+    >>> size = 5*u.arcsec
+
+Given this sky position and radius, we can use `~astropy.nddata.Cutout2D`
+in combination with ``use_fsspec=True`` and ``.section`` as follows:
+
+.. doctest-requires:: fsspec
+
+    >>> from astropy.nddata import Cutout2D
+    >>> from astropy.wcs import WCS
+    ...
+    >>> with fits.open(s3_uri, use_fsspec=True) as hdul:  # doctest: +REMOTE_DATA
+    ...     wcs = WCS(hdul[1].header)
+    ...     cutout = Cutout2D(hdul[1].section,  # use `.section` rather than `.data`!
+    ...                       position=position,
+    ...                       size=size,
+    ...                       wcs=wcs)
+
+As a final step, you can plot the cutout using Matplotlib as follows:
+
+.. doctest-requires:: fsspec
+
+    >>> import matplotlib.pyplot as plt
+    >>> from astropy.visualization import astropy_mpl_style
+    ...
+    >>> plt.style.use(astropy_mpl_style)  # doctest: +REMOTE_DATA
+    >>> plt.figure()  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    >>> plt.imshow(cutout.data, cmap='gray')  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    >>> plt.colorbar()  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
```

Comment:
I'm not quite sure how, but I think you can add the actual image here; now completely sure it is worth it, though.

Reply:
You mean, using the matplotlib plot directive? I am not actually sure if it plays well with pytest-doctestplus or not, but I agree would be nice if possible.